JS/cart.js

```diff
@@ -91,188 +61,80 @@ $(document).ready(function () {
+        document.getElementById("overlay_alert").classList.add("hidden");
+    });
+    //We update the sessionStorage and re-calculate the price of the article with the new quantity.
+    box_select_plus.addEventListener('click', function () {
         let elem = document.getElementById('quantity_id' + article.id);
         elem.value = parseInt(elem.value) + 1;        
         total_price = total_price - parseFloat(price.textContent) + article.price * elem.value;
         price.nodeValue = (article.price * elem.value).toFixed(2) + "€";
         document.getElementById("total_to_pay").textContent = "Total : " + total_price.toFixed(2) + " €";
         update_json_file(article, elem);
-    };
-
- /*
-there is a lot of call to JSON.parse and JSON.stringify, without counting the numberof parseInt|Float. 
-The cart should get a class with an internal storage dumped when necessary this will remove many excessive call to those functions. 
-This will also reduce memory movement making the job of the GC easier. keeping additional instances of the DOM object's could reduce the amount of lookup done in it and will enable you to create generic handler that will be instantiated only once.
-You may also want to check but JSON.parse should automatically parse integer and floats.
-*/
-    /*
-    * We make sure that the quantity of the article after the action is still superior to 0.
-    * We update the sessionStorage with the new quantity.
-    * We re-calculate the price of the article according to the new quantity
-    */
-    box_select_minus.onclick = function () {
+    });
+    //We ensure that the quantity is still > 0 & update sessionStorage with the new quantity & calculate the price of the article.
+    box_select_minus.addEventListener('click', function () {
         let elem = document.getElementById('quantity_id' + article.id);
         elem.value = parseInt(elem.value);
-        if (!(elem.value - 1 == 0)) {
+        if (elem.value - 1 != 0) {
             elem.value = elem.value - 1;
-            total_price = total_price - parseFloat(price.textContent) + article.price * (elem.value);
+            total_price = total_price - parseFloat(price.textContent) + article.price * elem.value;
             price.nodeValue = (article.price * elem.value).toFixed(2) + "€";
-            document.getElementById("total_to_pay").textContent = "Total : " + total_price.toFixed(2) + " €";   
+            document.getElementById("total_to_pay").textContent = "Total : " + total_price.toFixed(2) + " €";
             update_json_file(article, elem);
         }
-    };
-
-    /*
-   * We update the sessionStorage with the new quantity.
-   * We re-calculate the price of the article according to the new quantity
-   */
-    box_select_value.oninput = function () {      
+    });
+    //We update the sessionStorage and re-calculate the price of the article with the new quantity.
+    box_select_value.addEventListener('input', function () {
```

Comment:
Those event listening should be done inside of the cart_article class it is a functionality of the article. the class should expose an event interface to trigger when something change the interface should be made in a way allowing to change how the article is structured in the HTML